dowhy/causal_refuters/add_unobserved_common_cause.py

```diff
@@ -119,634 +123,821 @@ def __init__(self, *args, **kwargs):
-                    disable=not show_progress_bar,
-                    desc="Refuting Estimates: ",
-                ):
-                    for j in range(len(self.kappa_y)):
-                        new_data = self.include_confounders_effect(orig_data, self.kappa_t[i], self.kappa_y[j])
-                        new_estimator = CausalEstimator.get_estimator_object(
-                            new_data, self._target_estimand, self._estimate
-                        )
-                        new_effect = new_estimator.estimate_effect()
-                        refute = CausalRefutation(
-                            self._estimate.value,
-                            new_effect.value,
-                            refutation_type="Refute: Add an Unobserved Common Cause",
-                        )
-                        results_matrix[i][j] = refute.new_effect  # Populate the results
-
-                refute.new_effect_array = results_matrix
-                refute.new_effect = (np.min(results_matrix), np.max(results_matrix))
-                # Store the values into the refute object
-                refute.add_refuter(self)
-                if self.plotmethod is None:
-                    return refute
-
-                import matplotlib
-                import matplotlib.pyplot as plt
-
-                fig = plt.figure(figsize=(6, 5))
-                left, bottom, width, height = 0.1, 0.1, 0.8, 0.8
-                ax = fig.add_axes([left, bottom, width, height])
-
-                oe = self._estimate.value
-                contour_levels = [oe / 4.0, oe / 2.0, (3.0 / 4) * oe, oe]
-                contour_levels.extend([0, np.min(results_matrix), np.max(results_matrix)])
-                if self.plotmethod == "contour":
-                    cp = plt.contourf(self.kappa_y, self.kappa_t, results_matrix, levels=sorted(contour_levels))
-                    # Adding a label on the contour line for the original estimate
-                    fmt = {}
-                    trueeffect_index = np.where(cp.levels == oe)[0][0]
-                    fmt[cp.levels[trueeffect_index]] = "Estimated Effect"
-                    # Label every other level using strings
-                    plt.clabel(cp, [cp.levels[trueeffect_index]], inline=True, fmt=fmt)
-                    plt.colorbar(cp)
-                elif self.plotmethod == "colormesh":
-                    cp = plt.pcolormesh(self.kappa_y, self.kappa_t, results_matrix, shading="nearest")
-                    plt.colorbar(cp, ticks=contour_levels)
-                ax.yaxis.set_ticks(self.kappa_t)
-                ax.xaxis.set_ticks(self.kappa_y)
-                plt.xticks(rotation=45)
-                ax.set_title("Effect of Unobserved Common Cause")
-                ax.set_ylabel("Value of Linear Constant on Treatment")
-                ax.set_xlabel("Value of Linear Constant on Outcome")
-                plt.show()
 
-                return refute
+def _infer_default_kappa_t(
```

Comment:
what would be a good naming convention? shall we keep the "_" prefix for non-user facing functions? Or is there a more pythonic way of doing this?

Reply:
I'm adding the `_` everywhere I see a non-public function/method, also I'm adding the public facing functions into the `__init__.py` file of the module. I guess that's the pythonic way, I don't think there's a way to prevent the user from importing/using those functions.

Reply:
This sound good. We can follow this policy throughout.